core/amber/src/main/scala/edu/uci/ics/texera/web/resource/aiassistant/AiAssistantResource.scala

```diff
@@ -0,0 +1,132 @@
+package edu.uci.ics.texera.web.resource
+
+import edu.uci.ics.texera.web.auth.SessionUser
+import edu.uci.ics.texera.web.resource.aiassistant.AiAssistantManager
+import io.dropwizard.auth.Auth
+
+import javax.annotation.security.RolesAllowed
+import javax.ws.rs._
+
+import javax.ws.rs.core.Response
+import java.util.Base64
+import scala.sys.process._
+import java.util.logging.Logger
+
+@Path("/aiassistant")
+class AiAssistantResource {
+  private val logger = Logger.getLogger(classOf[AiAssistantResource].getName)
+
+  final private lazy val isEnabled = AiAssistantManager.validAIAssistant
+
+  @GET
+  @RolesAllowed(Array("REGULAR", "ADMIN"))
+  @Path("/isenabled")
+  def isAiAssistantEnabled: Boolean = isEnabled
+
+  /**
+    * To get the type annotation suggestion from OpenAI
+    */
+  @POST
+  @RolesAllowed(Array("REGULAR", "ADMIN"))
+  @Path("/getresult")
+  def getAiResponse(prompt: String, @Auth user: SessionUser): Response = {
+    val finalPrompt = prompt.replace("\\", "\\\\").replace("\"", "\\\"")
+    val requestBody =
+      s"""
+         |{
+         |  "model": "gpt-4",
+         |  "messages": [{"role": "user", "content": "$finalPrompt"}],
+         |  "max_tokens": 15
+         |}
+            """.stripMargin
+
+    try {
+      val url = new java.net.URL("https://api.openai.com/v1/chat/completions")
```

Comment:
For both "https://api.openai.com/v1/XXX" url, create a constant in format "https://api.openai.com/v1/", and each time when we do API call, we can use URL in this format: ${url}/chat/completions